2-mandatory/2-financial-times.js

```diff
@@ -14,24 +22,59 @@ function potentialHeadlines(allArticleTitles) {
     (you can assume words will always be seperated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
-}
 
+    let smallestTitle = allArticleTitles[0];
+
+    for (i = 1; i < allArticleTitles.length; i++) {
+
+        if (smallestTitle.split(" ").length > allArticleTitles[i].split(" ").length) {
+            smallestTitle = allArticleTitles[i];
+        }
+    }
+   return smallestTitle;
+}
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
+
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+    let headlinesWithNumbers = [ ];
+    const numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+    for (i = 0; i < allArticleTitles.length; i++) {
+        if(numbers.some(element => allArticleTitles[i].includes(element))) {
+            headlinesWithNumbers.push(allArticleTitles[i]);
+        }
+    }
+    return headlinesWithNumbers;
 }
 
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
 function averageNumberOfCharacters(allArticleTitles) {
-    // TODO
+    // To store the total characters per title
+    let totalCharactersPerTitle = [ ];
+
+    // To count the characters per title and add them to totalCharactersPerTitle
+    for (i = 0; i < allArticleTitles.length; i++) {  
```

Comment:
Issue: The iterator variable `i` in a for loop should be declared with `let`.
Suggestion: Use a for...of loop instead of a for loop to improve readability